docs/source/user_guide_pipeline.rst

```diff
@@ -133,6 +133,36 @@ can be added to the visualization by setting `hide_unused_outputs` to `False`:
     webbrowser.open("pipeline_full.html")
 
 
+*************************
+Pipeline State Management
+*************************
+
+Pipelines support checkpointing and resumption through state management features:
+
+.. code:: python
+
+    # Run pipeline until a specific component
+    state = await pipeline.run_until(data, stop_after="component_name", state_file="state.json")
```

Comment:
If we are returning the state, I think it would be cleaner let to the user save it to file or not, I'm not sure adding the `state_file` option is helpful.